lib/client/session.go

```diff
@@ -721,12 +720,16 @@ func (ns *NodeSession) pipeInOut(ctx context.Context, shell io.ReadWriteCloser,
 					fmt.Printf("\n\rError while sending force termination request: %v\n\r", err.Error())
 				}
 			})
+
+			// Force disconnect the session. We want to release the local terminal
+			// connected to the session rather than wait for the session to end.
+			ns.forceDisconnect.Store(true)
```

Comment:
Should this apply to both `SessionObserverMode` and `SessionModeratorMode`?

Reply:
When you press `Ctrl+C` you want to and the session on the client side and let the backend decide whether or not the session should be terminated for everyone else. In my opinion, it should be applied in both cases. Otherwise, the session will "hang" on the client side until the server session ends (as this happens currently).

@zmb3 @xacrimon Correct me if I'm wrong.

Reply:
Yep, that matches my understanding.

Reply:
Yep. This sounds correct to me.